### PR TITLE
nanomsg: fix pkgconfig file

### LINKED ISSUE #181812: nanomsg: Malformed pkg-config file
### Describe the bug
Upon inspection of the built  `nanomsg.pc` file in the package, I see that both `prefix` and `libdir` start with an absolute Nix path. This makes the library unusable through `pkg-config`, required for example by [Criterion](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/libraries/criterion/default.nix).

### Steps To Reproduce
1. Build project which requires `nanomsg` through `pkg-config`
2. It tries to include `/nix/store/xx-nanomsg//nix/store/xx-nanomsg/lib` and fails.

### Expected behavior
It includes `/nix/store/xx-nanomsg/lib` and builds correctly.

### Notify maintainers
No maintainer field in package file, git blame is full of broad mainenance commits.

### Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 5.18.9-arch1-1, Arch Linux, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.9.1`
 - channels(karim): `"git-goodies, nixgl, nixpkgs"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/home/karim/.nix-defexpr/channels/nixpkgs`

### PR DESCRIPTION
**Pull request by Thesola10**
###### Description of changes

Fixed #181812 by making the library path variable absolute, so as to not end up with a double prefix.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/nanomsg/pkgconfig.in.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/pkgconfig.in b/src/pkgconfig.in
+index d4c37522..35af803a 100644
+--- a/src/pkgconfig.in
++++ b/src/pkgconfig.in
+@@ -1,7 +1,7 @@
+ prefix=@CMAKE_INSTALL_PREFIX@
+ exec_prefix=${prefix}
+ includedir=${prefix}/include
+-libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
++libdir=@CMAKE_INSTALL_LIBDIR@
```
Comment (bjornfor) [inline pkgs/development/libraries/nanomsg/pkgconfig.in.patch:10]: If you make that `@CMAKE_INSTALL_FULL_LIBDIR@` you can submit it upstream.

Comment (Thesola10) [reply pkgs/development/libraries/nanomsg/pkgconfig.in.patch:10]: Thanks, I'll file it separately then remove this patch once upstream merges. Keeps it in sync with #181821

Comment (Thesola10) [reply pkgs/development/libraries/nanomsg/pkgconfig.in.patch:10]: Filed nanomsg/nanomsg#1085, now we wait.

**Comment by smancill** [comment]
Isn't it better to merge now and then replace the patch with upstream's?

It doesn't look they are in a hurry to reply.

**Comment by smancill** [comment]
I think it would be good if you use `fetchpatch` to download https://github.com/nanomsg/nanomsg/commit/e3323f19579529d272cb1d55bd6b653c4f34c064.patch (with a comment pointing to the PR) instead of manually including the patch, so we can get this merged.

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 181813` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>criterion</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>nanomsg</li>
    <li>python310Packages.nanomsg-python</li>
    <li>python39Packages.nanomsg-python</li>
  </ul>
</details>

**Comment by smancill** [comment]
One last thing, the commit message would be better as "fix pkgconfig file"

**Review by smancill — APPROVED** [review]

**Comment by Thesola10** [comment]
Upstream was merged right when this was merged T-T

**Comment by smancill** [comment]
That's why I asked you to switch to `fetchpatch` beforehand, so it's fine. We are using upstream commit.

**Pull request merged**